From 9230bf92101ee5167c373cc33fc0c2d5c961f333 Mon Sep 17 00:00:00 2001
From: Pascal Paillet <p.paillet@foss.st.com>
Date: Tue, 19 Nov 2024 10:44:21 +0100
Subject: [PATCH] arm64: dts: st: describe power domain for EXTI on stm32mp211

Set EXTI power domain to RET_PD to avoid entering in LPLV_Stop1
while a wake-up source is set on the CLUSTER domain.

Change-Id: I370017a5df751ba0b293ce2b2acb5df6d0464ec6
Signed-off-by: Pascal Paillet <p.paillet@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/419580
Tested-by: Francois CHOLEAU <francois.choleau@st.com>
Reviewed-by: Francois CHOLEAU <francois.choleau@st.com>
Domain-Review: Francois CHOLEAU <francois.choleau@st.com>
---
 arch/arm64/boot/dts/st/stm32mp211.dtsi | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/arm64/boot/dts/st/stm32mp211.dtsi b/arch/arm64/boot/dts/st/stm32mp211.dtsi
index 1f7b8344ce2f..3ff5b6feac11 100644
--- a/arch/arm64/boot/dts/st/stm32mp211.dtsi
+++ b/arch/arm64/boot/dts/st/stm32mp211.dtsi
@@ -2038,6 +2038,7 @@ exti1: interrupt-controller@44220000 {
 			interrupt-controller;
 			#interrupt-cells = <2>;
 			reg = <0x44220000 0x0 0x400>;
+			power-domains = <&RET_PD>;
 			interrupts-extended =
 				<&intc GIC_SPI 268 IRQ_TYPE_LEVEL_HIGH>,	/* EXTI_0 */
 				<&intc GIC_SPI 269 IRQ_TYPE_LEVEL_HIGH>,
-- 
2.50.1

